icu4c/source/common/mlbe.cpp

```diff
@@ -144,96 +131,87 @@ int32_t MlBreakEngine::divideUpRange(UText *inText, int32_t rangeStart, int32_t
 
 void MlBreakEngine::evaluateBreakpoint(UChar32* elementList, int32_t index, int32_t &numBreaks,
                                          UVector32 &boundary, UErrorCode &status) const {
-    char16_t featureList[MAX_FEATURE][MAX_FEATURE_LENGTH];
     if (U_FAILURE(status)) {
         return;
     }
 
-    UChar32 arr[4] = {-1, -1, -1, -1};
-    int32_t length = 0, listLength = 0;
-
-    const UChar32 w1 = elementList[0];
-    const UChar32 w2 = elementList[1];
-    const UChar32 w3 = elementList[2];
-    const UChar32 w4 = elementList[3];
-    const UChar32 w5 = elementList[4];
-    const UChar32 w6 = elementList[5];
-
-    length = 1;
-    if (w1 != INVALID) {
-        arr[0] = w1;
-        concatChar(u"UW1:", arr, length, featureList[listLength++], status);
+    UnicodeString featureList[MAX_FEATURE];
```

Comment:
why do we need to allocate MAX_FEATURE UnicodeString on the stack and then later on go through a loop to calculate the score by looping through it. How about just allocate one as feature and reuse it?
```
UnicodeString feature;
int32_t score = fNegativeSum;

if (elementList[0] != INVALID) {
    // Because when the key doesn't exist geti(key) returns 0  and 2 * 0 = 0,
    // the following line will not be changed, so we can skip checking fModel
    // includes key featureList[j] or not.
    score += (2 * fModel.geti(feature.setTo(u"UW1:", 4).append(elementList[0])));
}
 if (elementList[1] != INVALID) {
    score += (2 * fModel.geti(feature.setTo(u"UW2:", 4).append(elementList[1])));
}
.... and similar to the rest of if
```

Reply:
Done. Thanks